Library/Homebrew/cmd/vendor-install.sh

```diff
@@ -14,16 +14,15 @@ VENDOR_DIR="${HOMEBREW_LIBRARY}/Homebrew/vendor"
 # Built from https://github.com/Homebrew/homebrew-portable-ruby.
 if [[ -n "${HOMEBREW_MACOS}" ]]
 then
-  if [[ "${HOMEBREW_PROCESSOR}" == "Intel" ]] ||
+  if [[ "$(sysctl -n machdep.cpu.brand_string)" != "Apple"* ]] ||
```

Comment:
We can replace the original conditional entirely since this is true only if `${HOMEBREW_PROCESSOR} == "Intel"`.

Reply:
Also, for reference:
```
brew@HMBRW-A-001-M1-001 ~ % arch -x86_64 sysctl -a machdep.cpu.brand_string
machdep.cpu.brand_string: Apple M1
```

Reply:
I don't really like this check, at least not here, because it's checking the processor name rather than the architecture. Feels to me like `brew.sh` should expose some more of https://github.com/Homebrew/brew/blob/98b33e673200778b06f564bba26b60be7918e818/Library/Homebrew/brew.sh#L20-L23 as variables we can check here and have the appropriate conditionals.

Reply:
I did have a look there first to see if there's something to be reused, and I'm not sure there is without having to check several of these, because:
```
brew@HMBRW-A-001-M1-001 ~ % arch -x86_64 sysctl -a hw.optional.arm64
hw.optional.arm64: 1
brew@HMBRW-A-001-M1-001 ~ % arch -x86_64 sysctl -a hw.optional.x86_64
hw.optional.x86_64: 1
```
Could you elaborate why checking the processor name isn't a good idea here? It is gated behind `HOMEBREW_MACOS`, so I think the implicit assumptions about the mapping between the processor name and architecture are correct, and unlikely to change for a long time.

Reply:
> I did have a look there first to see if there's something to be reused, and I'm not sure there is without having to check several of these

I think it's OK to check several of these, they seem super quick to run.

> Could you elaborate why checking the processor name isn't a good idea here? It is gated behind `HOMEBREW_MACOS`, so I think the implicit assumptions about the mapping between the processor name and architecture are correct, and unlikely to change for a long time.

1. it feels like a bit of a hack. The name implies architecture but if we can check the architecture instead: we should
2. once people start doing ARM Hackintoshes: it'll break. We don't support them but they will complain and waste our time anyway.

Reply:
Ok, done.